bgpd/bgp_ovsdb_route.c

```diff
@@ -417,7 +417,7 @@ bgp_ovsdb_lookup_local_nexthop(char *ip)
     OVSREC_BGP_NEXTHOP_FOR_EACH(row, idl) {
         if (row->ip_address)
         {
-            if (strcmp(ip, row->ip_address) == 0) {
+            if (ip_addr_cmp(ip, row->ip_address)) {
```

Comment:
positive or non zero?
if you want to use bool result, than you should use something like "is_equal". If you extend "cmp" functions family, you should use cmp semantic